### PR TITLE
feat: expose tab bar height

### PR DESCRIPTION
**Pull request by samchamberland**
### Motivation

Some user(s) mentioned their use case of wanting to implement a transparent tab bar. This is do-able by absolutely positioning the tab bar in `tabBarOptions.style` - However, when using this approach, it'd be ideal to have a way to access the tab bar height (e.g. to add padding if needed). This PR exposes the tab bar height as part of the public API via context.

Closes https://github.com/react-navigation/tabs/issues/184
Also somewhat related to https://github.com/react-navigation/tabs/issues/46

### Test plan

A new example of a context aware transparent (absolutely positioned) tab bar was added to the `example` directory. 

<details>
<summary>GIF of example</summary>

ssttb

</details>

### CODE REVIEW

File: src/views/BottomTabBar.tsx [hunk 1]
```diff
@@ -20,6 +20,10 @@ import {
   KeyboardAnimationConfig,
 } from '../types';
 
+type Props = {
+  onHeightChange?: (height: number) => void;
```
Comment (satya164) [inline src/views/BottomTabBar.tsx]: Can you move this to `BottomTabBarProps`?

File: src/views/BottomTabBar.tsx [hunk 2]
```diff
@@ -198,12 +202,19 @@ class TabBarBottom extends React.Component<BottomTabBarProps, State> {
...
-        width,
+    this.setState(
+      {
+        layout: {
+          height,
+          width,
+        },
       },
-    });
+      () => {
```
Comment (satya164) [inline src/views/BottomTabBar.tsx]: Is it necessary to wait for the setState callback for this? Something like this should work fine:

```js
this.props.onHeightChange && this.props.onHeightChange(...)
this.setState(...)
```

File: src/navigators/createBottomTabNavigator.tsx [hunk 3]
```diff
@@ -144,27 +154,29 @@ class TabNavigationView extends React.PureComponent<Props, State> {
...
+                  key={route.key}
+                  style={StyleSheet.absoluteFill}
+                  isVisible={isFocused}
+                >
+                  {renderScene({ route })}
+                </ResourceSavingScene>
+              );
+            })}
+          </ScreenContainer>
+          {this._renderTabBar()}
```
Comment (satya164) [inline src/navigators/createBottomTabNavigator.tsx]: I think this should not be under the context provider

**Comment by samchamberland** [comment]
Updated the PR. Thanks for looking at this.

**Pull request closed without merging**